### PR TITLE
Xref parser mgiparser_ccds

### PR DESCRIPTION
**Pull request by premanand17**
## Requirements

- Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion;
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/release/90/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/release/90/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

Updates:

Used Text:CSV for parsing the file
Tidy up

## Use case

XrefParser for source MGI (MGI_CCDS) for mouse (mus musculus)

## Benefits

Code quality improvements.

## Possible Drawbacks

None

## Testing

No until test. However, have tested it by running the xref_parser script and checked the _xref database for row counts.

### CODE REVIEW

File: misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm [hunk 1]
```diff
@@ -23,126 +24,145 @@ use strict;
...
-  my $sql = 'select source_id, priority_description from source where name like "MGI"';
+  my $sql = 'select source_id from source where name like "MGI"';
   my $sth = $dbi->prepare($sql);
 
   $sth->execute();
-  my ($mgi_source_id, $desc);
-  $sth->bind_columns(\$mgi_source_id, \$desc);
+  my ($mgi_source_id);
+  $sth->bind_columns( \$mgi_source_id );
   my @arr;
```
Comment (nerdstrike) [inline misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm]: Could you change this variable name to something useful? Like @source_ids or something.

File: misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm [hunk 2]
```diff
@@ -23,126 +24,145 @@ use strict;
...
+'select x.label, x.xref_id from xref x, source s where x.source_id = s.source_id and s.name ="CCDS"';
+
   my %ccds_label_to_xref_id;
   $sth = $dbi->prepare($sql);
   $sth->execute();
   my ($xref_id);
-  $sth->bind_columns(\$lab, \$xref_id);
-  while (my @row = $sth->fetchrow_array()) {
-    $ccds_label_to_xref_id{$row[0]} = $row[1];
+  $sth->bind_columns( \$lab, \$xref_id );
```
Comment (nerdstrike) [inline misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm]: Binding variables but fetching to an array?

File: misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm [hunk 3]
```diff
@@ -23,126 +24,145 @@ use strict;
...
-  my $ccds_missing = 0;
+  my $count              = 0;
+  my $ccds_missing       = 0;
   my $entrezgene_missing = 0;
 
   my $mgi_io = $self->get_filehandle($file);
   if ( !defined $mgi_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
+    croak "ERROR: Could not open $file\n";
```
Comment (nerdstrike) [inline misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm]: As discussed in other pull requests, if we're generating an error, we don't need ERROR in the message.

File: misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm [hunk 4]
```diff
@@ -23,126 +24,145 @@ use strict;
...
+      sep_char           => "\t",
+      empty_is_undef     => 1,
+      strict             => 1,
+      allow_loose_quotes => 1,
+    }
+  ) or croak "Cannot use file $file: " . Text::CSV->error_diag();
+
+  $input_file->column_names(
+    [qw( chromosome g_accession gene gene_id ccds_id)] )
+    ;    #ignore the rest of the columns
```
Comment (nerdstrike) [inline misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm]: This is some eccentric perl-tidying

File: misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm [hunk 5]
```diff
@@ -23,126 +24,143 @@ use strict;
...
+  $input_file->column_names( [qw( chromosome g_accession gene gene_id ccds_id)] );    #ignore the rest of the columns
+
+  while ( my $data = $input_file->getline_hr($mgi_io) ) {
+    my $ccds      = $data->{'ccds_id'};
+    my $gene_name = $data->{'gene'};
+    if ( defined( $ccds_label_to_xref_id{$ccds} ) ) {
+      if (  defined( $accession{$gene_name} )
+        and defined( $label{ $accession{$gene_name} } ) )
+      {
+        my $acc = $accession{$gene_name};
```
Comment (tgrego) [inline misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm]: $acc is previously declared on line 72.
It's a good idea not to reuse variable names in the same lexical scope, consider renaming it.

**Comment by avullo** [comment]
Closing this as Prem has made an equivalent one not merged from master.

**Pull request closed without merging**